src/main/java/seedu/address/model/pet/ColorPattern.java

```diff
@@ -13,7 +13,7 @@ public class ColorPattern implements Comparable<ColorPattern> {
      * @param value The string representation of color pattern.
      */
     public ColorPattern(String value) {
-        if (value == null) {
+        if (value == null || !value.matches("^[a-zA-Z0-9\\s]+$")) {
```

Comment:
Also, does a string with whitespaces match this regex?

Reply:
this one yes